Component flaky test

<!--
Thanks for submitting a bug report. Please fill the template below,
otherwise we will not be able to process this bug report.
-->

#### Summary
<!-- Summarize the problem in a few sentences: -->

Component flaky test

#### Steps to Reproduce
<!-- How can we reproduce the problem? -->

1. Run CI
2. Flip a coin

<!--
Please upload relevant configuration (as .txt).
If you use the command "ttn-lw-stack config", you can redact sensitive config.
-->

#### What do you see now?
<!--
Please paste terminal output, upload logs (as .txt) or upload screenshots.
-->

```
  WARN No cookie hash key configured, generated a random one hash_key=93CCC41C208F9FC51C8E518AE36138D01D8097BD4ECC920E5802F45B107F11D67AA7425396882FF025F9CA83534FFA35063031D9BC4CA6BA4E8A253D94A0A963 namespace=web test_name=TestRun
  WARN No cookie block key configured, generated a random one block_key=6B1DF7EA88697828BD4ECF44C1481D63F26AC6B1B623BE272AA1431017364BC2 namespace=web test_name=TestRun
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestRun
  WARN No rights TTL configured                 test_name=TestRun
  WARN No cluster key configured, generated a random one key=7d78100fa80ef3bfbfea4b318f413b6d116aca0f0c490433f524e52f2f2bfb1f test_name=TestRun
  INFO Received signal, exiting...              signal=test-signal-string test_name=TestRun
  WARN No cookie hash key configured, generated a random one hash_key=86687249D2222A107C32CEBC93D6DD3E388B489BB1AF4EC160C9AFC2161A6A3F9FED3B6F8A6ED309E41C802058FD1124C6E240B54FE4C66D38775D0811B29A6B namespace=web test_name=TestTasks
  WARN No cookie block key configured, generated a random one block_key=CB9D6DBB721C1793F9C59B3FE8DB48BE9282CBC2D40B70AA1F76924CADBF67E4 namespace=web test_name=TestTasks
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestTasks
  WARN No rights TTL configured                 test_name=TestTasks
  WARN No cluster key configured, generated a random one key=61185c15777c3b820e54a99c0b32ad0ca867854603db0220e2af42c41d25747c test_name=TestTasks
  WARN transport: http2Server.HandleStreams failed to read frame: io: read/write on closed pipe namespace=grpc test_name=TestTasks
  WARN No cookie hash key configured, generated a random one hash_key=B0C9BB24D8D9A0F725CD6249F77034715F308EF025ACBD44561D1F70DBBFFA010A67C1B221DFF22D4252B54F7044FF645717BB209E9EFBFD119393E396BB30BE namespace=web test_name=TestPProf
  WARN No cookie block key configured, generated a random one block_key=0E4B9C0DEEEC30B3F1B959375CC29AF6286D04FAEEC9C941B309D43E57C83F3F namespace=web test_name=TestPProf
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestPProf
  WARN No rights TTL configured                 test_name=TestPProf
  INFO Listening for connections                address=0.0.0.0:8097 namespace=web protocol=Web test_name=TestPProf
  WARN No cookie hash key configured, generated a random one hash_key=EFCE141A03354A56FF8C060E1A91AEEE06FF8154FAEB1E1E5B79F1264316CBF9FF57CE37270E90300B92326B77370CF9F78DBF28CDB13C67A739159515531653 namespace=web test_name=TestPeers
  WARN No cookie block key configured, generated a random one block_key=93119D06BCFDA21D7918DD47B92BAAB9979FA3CA9D4A2D15FC6ACD87358ABC9E namespace=web test_name=TestPeers
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestPeers
  WARN No rights TTL configured                 test_name=TestPeers
  WARN No cluster key configured, generated a random one key=f7a24f779eb21f55ec8fb9ffd2e109b2ba9ec1c8a0ba54ea72a09cf68c386d68 test_name=TestPeers
  WARN transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:44393->127.0.0.1:48272: use of closed network connection namespace=grpc test_name=TestPeers
  WARN No cookie hash key configured, generated a random one hash_key=B082377670B55AA7810E0294A71B945BC08C6106C2B3BCC988B738CD3FCEF3757154F2AAB661E5CEE03665E4F852C2798CAFEDBB59503688C72A5F92ACC762C3 namespace=web test_name=TestNew
  WARN No cookie block key configured, generated a random one block_key=A4D74CAE05F1876A7AF0025317CC7F7F79B2DB7BE0B1DF0B522BE78ACD801659 namespace=web test_name=TestNew
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestNew
  WARN No rights TTL configured                 test_name=TestNew
  WARN No cookie hash key configured, generated a random one hash_key=113F8A5C09F046F30493D69AFEB3AF3039947E8418EC6A91B4C715D0A9A8C3B5EC8704ED7B47B889D19E5B58AD460A84459B0BDE441B16199DF7F7A8F99B749E namespace=web test_name=TestNew
  WARN No cookie block key configured, generated a random one block_key=D0322DA826B0BE5EDBC4072118DF1A46599D6E820277FB9DE55A2F8654167896 namespace=web test_name=TestNew
  WARN No static assets found in any search path namespace=web search_paths=[] test_name=TestNew
  WARN No rights TTL configured                 test_name=TestNew
--- FAIL: TestLogger (0.00s)
    doc.go:66: 
        /home/travis/gopath/src/go.thethings.network/lorawan-stack/pkg/component/component_test.go:67
        Expected collection to have length equal to [5], but it's length was [6] instead! contents: [0xc000274bc0 0xc000274c40 0xc000274f40 0xc000275040 0xc0004c6000 0xc000275200]
FAIL
coverage: 55.6% of statements
FAIL	go.thethings.network/lorawan-stack/pkg/component	0.192s
```

#### What do you want to see instead?
<!-- Please add some examples or mock-ups if applicable. -->

Test pass

#### Environment
<!--
Your environment: OS/Browser/Gateway/Device/...? Versions? IDs/EUIs?
Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable.
-->

Travis

#### How do you propose to implement this?
<!-- Please think about how this could be fixed. -->

Not sure

#### Can you do this yourself and submit a Pull Request?
<!-- You can also @mention experts if you need help with this. -->

Can probably sanity-check review

--- a/pkg/events/handler_test.go
+++ b/pkg/events/handler_test.go
@@ -17,6 +17,7 @@ package events_test
 import (
 	"context"
 	"fmt"
+	"runtime"
 	"testing"
 
 	"github.com/smartystreets/assertions"
@@ -68,6 +69,8 @@ func TestChannelReceive(t *testing.T) {
 	eventChan.Notify(events.New(test.Context(), "evt", nil, nil))
 	eventChan.Notify(events.New(test.Context(), "evt", nil, nil))
 	eventChan.Notify(events.New(test.Context(), "overflow", nil, nil))
+
+	runtime.Gosched()
 
 	ctx, cancel := context.WithCancel(test.Context())
 

--- a/pkg/rpcserver/rpcserver_test.go
+++ b/pkg/rpcserver/rpcserver_test.go
@@ -16,6 +16,7 @@ package rpcserver_test
 
 import (
 	"context"
+	"runtime"
 	"testing"
 
 	"github.com/gogo/protobuf/types"
@@ -105,6 +106,8 @@ func TestNewRPCServer(t *testing.T) {
 		})
 		a.So(mock.pushCtx.Value(&mockKey2{}), should.Resemble, "bar")
 
+		runtime.Gosched()
+
 		a.So(logHandler.entries, should.HaveLength, 1)
 	})
 
@@ -129,6 +132,8 @@ func TestNewRPCServer(t *testing.T) {
 			"grpc.request.foo":    "bar",
 		})
 		a.So(mock.subCtx.Value(&mockKey2{}), should.Resemble, "foo")
+
+		runtime.Gosched()
 
 		a.So(logHandler.entries, should.HaveLength, 2)
 	})

